lib/parse.js

```diff
@@ -2174,6 +2258,53 @@ function parse($TEXT, options) {
         });
     };
 
+    function usingdefs(no_in, kind) {
+        var using_defs = [];
+        var def;
+        for (;;) {
+            var sym_type =
+                kind === "using"
+                    ? AST_SymbolUsing
+                    : kind === "await using"
+                    ? AST_SymbolAwaitUsing
+                    : null;
+
+            def = new AST_UsingDef({
+                start: S.token,
+                name: as_symbol(sym_type),
+                value: is("operator", "=")
+                    ? (next(), expression(false, no_in))
+                    : !no_in 
+                    ? croak("Missing initializer in " + kind + " declaration")
+                    : null,
+                end: prev(),
+            });
+            if (def.name.name == "import") croak("Unexpected token: import");
+            using_defs.push(def);
+            if (!is("punc", ","))
+                break;
+            next();
+        }
+        return using_defs;
+    }
+
+    var using_ = function(no_in) {
+        return new AST_Using({
+            start       : prev(),
+            definitions : usingdefs(no_in, "using"),
+            end         : prev()
+        });
+    };
+
+    var await_using_ = function(no_in) {
+        // Assumption: When await_using_ is called, only the `await` token has been consumed.
```

Comment:
If you'd like, you could make this more obvious by accepting a `start` argument in this function, allowing you to skip both tokens before calling this. But this works fine too.